HabitTracker.ZubePierreBasali/HabitTracker.ZubePierreBasali/DataBase.cs

```diff
@@ -0,0 +1,952 @@
+            SqliteDataReader reader = command.ExecuteReader();
+            if (!reader.HasRows)
+            {
+                DisplayText("First time entering data for this record, please choose a unit:\n");
+                do
+                {
+                    unit = GetTable();
+                    if (string.IsNullOrEmpty(unit) || string.IsNullOrWhiteSpace(unit))
+                    {
+                        DisplayText("Please enter a valid unit for the record.", 10, 500);
+                        ClearText();
+                    }
+                } while (string.IsNullOrEmpty(unit) || string.IsNullOrWhiteSpace(unit));
+                connection.Close();
+                return unit;
+            }
+            else
+            {
+                reader.Read();
+                unit = reader["Unit"].ToString();
+                connection.Close();
+                return unit;
+            }
+        }
+
+        public static void ExecuteQuery(string connectionString, string sqlCommand, bool test = false, string date = "", string endDate = "", string unit = "", int quantity = 0, int id = -1, int endId = -1)
+        {
+            try
+            {
+                SqliteConnection connection = new(connectionString);
+                connection.Open();
+                SqliteCommand? tableCmd = connection.CreateCommand();
+                tableCmd.CommandText = sqlCommand;
+
+                // Parametrized query are used to avoid injections
+                tableCmd.Parameters.AddWithValue("$date", date);
+                tableCmd.Parameters.AddWithValue("$quantity", quantity);
+                tableCmd.Parameters.AddWithValue("$unit", unit);
+                tableCmd.Parameters.AddWithValue("$id", id);
+                tableCmd.Parameters.AddWithValue("$endDate", endDate);
+                tableCmd.Parameters.AddWithValue("$endId", endId);
+                tableCmd.ExecuteNonQuery();
+                connection.Close();
+            }
+            catch (Exception ex)
+            {
+                ClearText();
+                DisplayText($"{ex.Message}\n", 5, 2000);
+            }
+        }
+
+        private static string Redo(string message)
+        {
+            DisplayText(message);
+            do
+            {
+                readResult = Console.ReadLine();
+                if (readResult.ToLower() != "y" && readResult.ToLower() != "n")
+                {
+                    DisplayText("Please choose Between Yes (Y) or No (N).", 5, 500);
+                    ClearText();
+                }
+                else if (readResult.ToLower() == "n")
+                {
+                    DisplayText("Press Enter to continue.");
+                    Console.ReadLine();
+                }
+            } while (readResult.ToLower() != "n" && readResult.ToLower() != "y");
+            return readResult;
+        }
+
+        private static void DisplayText(string message, int speed = 5, int sleep = 50)
```

Comment:
This is... not ideal. 

Why not just do `Console.Write("Message")` or `Console.WriteLine("Message")`?

App hangs. `Thread.Sleep` is very very bad practice.